src/main/java/io/gingersnapproject/k8s/CacheRuleInformer.java

```diff
@@ -0,0 +1,224 @@
+package io.gingersnapproject.k8s;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.Map.Entry;
+import java.util.Optional;
+
+import javax.enterprise.event.Observes;
+import javax.enterprise.inject.Instance;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
+import io.gingersnapproject.cdc.DynamicRuleManagement;
+import io.gingersnapproject.cdc.configuration.Configuration;
+import io.gingersnapproject.cdc.configuration.Connector;
+import io.gingersnapproject.cdc.configuration.Rule;
+import io.gingersnapproject.k8s.configuration.KubernetesConfiguration;
+import io.gingersnapproject.proto.api.config.v1alpha1.EagerCacheRuleSpec;
+import io.gingersnapproject.proto.api.config.v1alpha1.KeyFormat;
+import io.gingersnapproject.proto.api.config.v1alpha1.EagerCacheRuleSpec.Builder;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.protobuf.InvalidProtocolBufferException;
+import com.google.protobuf.util.JsonFormat;
+
+import io.fabric8.kubernetes.api.model.ConfigMap;
+import io.fabric8.kubernetes.client.KubernetesClient;
+import io.fabric8.kubernetes.client.informers.ResourceEventHandler;
+import io.fabric8.kubernetes.client.informers.SharedIndexInformer;
+import io.quarkus.runtime.ShutdownEvent;
+import io.quarkus.runtime.StartupEvent;
+
+@Singleton
+public class CacheRuleInformer {
+
+   private static final Logger log = LoggerFactory.getLogger(CacheRuleInformer.class);
+
+   @Inject
+   Instance<KubernetesClient> client;
+   @Inject
+   KubernetesConfiguration configuration;
+   @Inject
+   DynamicRuleManagement drm;
+   @Inject
+   Configuration config;
+   private SharedIndexInformer<ConfigMap> informer;
+
+   void startWatching(@Observes StartupEvent ignore) {
+      if (client.isUnsatisfied() || configuration.configMapName().isEmpty()) {
+         log.info("Kubernetes client not found, not watching config map");
+         return;
+      }
+
+      String ruleName = configuration.configMapName().get();
+      log.info("Informer on rules in {}", ruleName);
+      final long RESYNC_PERIOD = 60 * 1000L;
```

Comment:
No need for this to be `final`. We can even use `var` now we're on java 17 :slightly_smiling_face: